src/QUICSocket.ts

```diff
@@ -63,13 +88,13 @@ class QUICSocket extends EventTarget {
     } catch (e) {
       // `BufferTooShort` and `InvalidPacket` means that this is not a QUIC
       // packet. If so, then we just ignore the packet.
-      if (e.message !== 'BufferTooShort' && e.message !== 'InvalidPacket') {
-        // Emit error if it is not a `BufferTooShort` or `InvalidPacket` error.
-        // This would indicate something went wrong in header parsing.
-        // This is not a critical error, but should be checked.
-        this.dispatchEvent(new events.QUICSocketErrorEvent({ detail: e }));
+      if (e.message === 'BufferTooShort' || e.message === 'InvalidPacket') {
+        return;
       }
-      return;
+      // If the error is niether `BufferTooShort` or `InvalidPacket`, this
+      // may indicate something went wrong in the header parsing, which should
+      // be a software error.
+      throw e;
```

Comment:
We absolutely should not throw here. Any invalid packets should be discarded otherwise the program **WILL** crash due to random internet traffic.

Reply:
That's not what the comment said.

Reply:
If the comment is wrong it should have been fixed. In which case all errors could just be ignored and there's no need to check the error message at all.

Reply:
I misread this, nevermind.

Reply:
It should be tested though, I've never tested it with random binary data.

Reply:
From what I recall, there is at least 1 test that throws random data through the socket, let me find it.

Yeah at `tests/QUICClient.test.ts:659` under `describe('handles random packets'...`